### PR TITLE
Add Color For Individual Cells

### PR DESCRIPTION
**Pull request by Mitsuwa**
@knz @mattn @hasit

# What

This adds a new ability `SetCellColor` to set the Cell Colors individually, while combined with/or without Column Colors.  Cell Colors with take PRIORITY over Column Colors (otherwise it wouldn't really do anything 😸 )

# Why

Currently if you set Column coloring the entire column will be color coded that way.  There are times where it would be really nice if we can maintain the color for the column but allow for individual cells to also to also take a different coloring.  Maybe we have a list of files in a table and we want to color code some of them based on certain status for example.

# Testing

I've tested this a few different ways, i've also added a section to the readme to show how to use these column cells.  The readme has the code for these generated tables.  I also tested what happens if the values of (col,row) passed to SetCellColors are out of range, that isn't an issue because `getColorForCell` just returns an emtpy string that is checked for

* NO COLUMN COLORS WITH CELL COLORS ( NOT MERGED )

image

* COLUMN COLORS WITH CELL PRIORITY ( NOT MERGED )

image

* NO COLUMN COLORS WITH CELL COLORS ( MERGED )

image

* COLUMN COLORS WITH CELL PRIORITY ( MERGED )

image

### CODE REVIEW

**Comment by Mitsuwa** [comment]
@mattn would you mind taking a look at this?

File: table.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ type Table struct {
 	borders        Border
 	colSize        int
 	headerParams   []string
+	cellParams     []CellColor
```
Comment (mattn) [inline table.go]: Could you please make new interface `CellParam` that just have method `param()` and type `[]CellParam` here. (FYI: struct CellColor should have `param()`)

**Comment by olekukonko** [comment]
I love the Idea but I honestly think this is just complex, I would have prefered a simpler approach such as

```
table.Append(v)  // to append plain 
table.Rich(v, tablewriter.Colors{}) // to rows with color attribute 
``` 

#### Reason 
- We already have something that works colon wide 
- Tracking row number might not be as easy as you think 

What do you guys feel?

**Comment by Mitsuwa** [comment]
Screen Shot 2019-08-29 at 3 14 21 PM

@olekukonko @mattn 

Hey folks, Sorry for the delay on updating this pr!

Please find the updartes above, I believe this will fit your needs that @olekukonko was mentioning and simplifying to use a standard Rich append line method

**Comment by Mitsuwa** [comment]
Screen Shot 2019-08-29 at 3 21 28 PM

You can also see that the unmerged cells functions appropriately as well

**Comment by Mitsuwa** [comment]
I also pushed a small fix to fix the current tests.

**Comment by mattn** [comment]
LGTM

**Comment by mattn** [comment]
Thanks. I'll tag soon.
